src/main/java/org/altbeacon/beacon/service/scanner/CycledLeScanner.java

```diff
@@ -193,14 +223,14 @@ public void run() {
 
     protected abstract void startScan();
 
-    @SuppressLint("NewApi")
+    @MainThread
     protected void scanLeDevice(final Boolean enable) {
         try {
             mScanCyclerStarted = true;
             if (getBluetoothAdapter() == null) {
                 LogManager.e(TAG, "No Bluetooth adapter.  beaconService cannot scan.");
             }
-            if (enable) {
+            if (enable && mScanningEnabled) {
```

Comment:
Not 100% sure why we would not want to execute this block if mScanningEnabled is false